control/control.go

```
@@ -226,6 +262,62 @@ func New(cfg *Config) *pluginControl {
 	return c
 }
 
+func (p *pluginControl) HandleGomitEvent(e gomit.Event) {
+	switch v := e.Body.(type) {
+	case *control_event.LoadPluginEvent:
+		// while task is running we care about collectors being loaded
+		// only collectors can affect current workflow by changing metric catalog
+		if core.PluginType(v.Type) == core.CollectorPluginType {
+			for taskID, td := range p.taskView {
+				// get all new metrics and collector plugins
+				newMetrics, newPlugins, _ := p.getMetricsAndCollectors(td.requested, td.configTree)
+
+				// RequestedMetrics are only required while starting a task
+				// at this stage we already know it (part of task data)
+				errs := p.SubscribeDeps(taskID, newMetrics, []core.RequestedMetric{}, newPlugins, td.configTree)
+				if len(errs) > 0 {
+					log.WithFields(log.Fields{
+						"_block":         "control",
+						"event":          v.Namespace(),
+						"plugin-name":    v.Name,
+						"plugin-version": v.Version,
+						"plugin-type":    core.PluginType(v.Type).String(),
+					}).Error("error subscribing dependencies")
+					p.UnsubscribeDeps(taskID, newMetrics, newPlugins)
+					continue
+				}
+			}
+		}
+	case *control_event.UnloadPluginEvent:
+		// while task is running we care about collectors being unloaded
+		// only collectors can affect current workflow by changing metric catalog
+		if core.PluginType(v.Type) == core.CollectorPluginType {
+			for taskID, td := range p.taskView {
+				newMetrics, newPlugins, _ := p.getMetricsAndCollectors(td.requested, td.configTree)
```

Are we swallowing an error here?